packages/danger/src/AndroidPermissionsChecker.ts

```diff
@@ -0,0 +1,92 @@
+ * - Check if a new `use-permissions` tag was added.
+ * - Warn about also adding new permissions to the XDL blacklist.
+ * - _Never_ fail CI, but add a warning to the PR as a reminder.
+ */
+export async function checkAndroidPermissions(): Promise<void> {
+  const manifests = getTouchedManifests();
+  if (manifests.length === 0) {
+    console.log('Everything is ok 🎉');
+    return;
+  }
+
+  const results = await Promise.all(
+    manifests.map(async manifest => ({
+      manifest,
+      permissions: await getAddedManifestPermissions(manifest),
+    }))
+  );
+
+  outputResult(results.filter(result => result.permissions.length > 0));
+}
+
+/**
+ * Get a list of modified or created Android manifests, from danger's git.
+ * This only returns `AndroidManifest.xml` files within the `packages/` folder.
+ */
+function getTouchedManifests(): string[] {
+  const touchedFiles = [...danger.git.modified_files, ...danger.git.created_files];
+  return uniq(
+    touchedFiles.filter(file => file.startsWith('packages') && file.endsWith('AndroidManifest.xml'))
+  );
+}
+
+/**
+ * Determine if the Android manifest file has an added `use-permission` tag.
+ */
+async function getAddedManifestPermissions(filePath: string): Promise<string[]> {
+  const result = await danger.git.diffForFile(filePath);
+  if (!result) {
+    return [];
+  }
+
+  return result.diff
+    .split('\n')
+    .map(line => line.match(TOUCHED_PERMISSION_REGEX)?.[1]?.trim() || '')
+    .filter(Boolean);
+}
+
+/**
+ * Generate a report and output a message warning for the added permissions in the Android manifest.
+ */
+function outputResult(results: { manifest: string; permissions: string[] }[]): void {
+  if (results.length === 0) {
+    return;
+  }
+
+  const info = results.map(
+    result =>
+      `<code>${danger.github.utils.fileLinks([result.manifest], false)}</code>
+${result.permissions.map(permission => `\\- \`${permission}\``).join('\n')}`
+  );
+
+  warn(
+    `**Manifests with new permissions**
+------
+${info.join('<br /><br />')}
+`
+  );
+
+  markdown(
+    `### ⚠️  Android permissions added
+Please make sure these permissions are absolutely required for the core functionality of the modules. All permissions added in the Android Manifests are included, by default, in the standalone builds.
+
+<details>
+  <summary>What can I do?</summary>
+
+  - Keep it in when you are confident it's required for the primary functionality of the API, and add it to [the XDL schema](https://github.com/expo/universe/blob/5c84d1b6e82ba3eca483170be72deee49d29d916/server/www/xdl-schemas/UNVERSIONED-schema.json#L686).
+  - Reconsider if the permission is required, and remove it from the \`AndroidManifest.xml\`.
+  - Add the permission to the [XDL blacklist](https://github.com/expo/expo-cli/blob/d75089b2e9f11b36f936967313d847fcc45f4e76/packages/xdl/src/detach/AndroidShellApp.js#L780) to allow users to opt-out of the permissions, by excluding it from the \`android.permissions\` manifest property.
```

Comment:
this is pretty small, but when reading this it wasn't clear to me whether I should do just one of these, or all of them? Like if I decide to **keep the change**, I should probably do _both_ (1) and (3)? aka
```
  - Keep it in when you are confident it's required for the primary functionality of the API, and add it to [the XDL schema](https://github.com/expo/universe/blob/5c84d1b6e82ba3eca483170be72deee49d29d916/server/www/xdl-schemas/UNVERSIONED-schema.json#L686).
  - Add the permission to the [XDL blacklist](https://github.com/expo/expo-cli/blob/d75089b2e9f11b36f936967313d847fcc45f4e76/packages/xdl/src/detach/AndroidShellApp.js#L780) to allow users to opt-out of the permissions, by excluding it from the \`android.permissions\` manifest property.
```

If that's the case I would suggest indenting (3) under (1). So format would be:

```suggestion
  - Reconsider if the permission is required, and remove it from the \`AndroidManifest.xml\`.
  _or_
  - Keep it in when you are confident it's required for the primary functionality of the API, and add it to [the XDL schema](https://github.com/expo/universe/blob/5c84d1b6e82ba3eca483170be72deee49d29d916/server/www/xdl-schemas/UNVERSIONED-schema.json#L686).
    - Make sure to add the permission to the [XDL blacklist](https://github.com/expo/expo-cli/blob/d75089b2e9f11b36f936967313d847fcc45f4e76/packages/xdl/src/detach/AndroidShellApp.js#L780) to allow users to opt-out of the permissions, by excluding it from the \`android.permissions\` manifest property.
```

Reply:
Ah yeah, good point! Originally I had 2 points, where I used the `-` sign because it was either 1 of them. But adding the 3rd made it more complex. Let's go for that one!

As for the `expo-screen-capture` permission, no we definitely need the user's location to prevent screen captures 😘  no jk, it's gone before merging it! I'll remove the commit once we are done, but when I do that it will remove the message as well.